extra/resources/ifspeed

```diff
@@ -165,13 +182,36 @@ monitor() {
     return ${ret}
 }
 
+# This function tries to guess nic interface by IP
+get_nic_name_by_ip(){
+    # $FINDIF takes its parameters from the environment
+    NICINFO=`${FINDIF}`
+    rc=$?
+    if [ $rc -eq 0 ];then
+        NICINFO=`echo "${NICINFO}" | sed -e 's/netmask\ //;s/broadcast\ //'`
+        echo "${NICINFO}" | cut -d" " -f1
+    else
+        echo ""
+    fi
+}
+
 validate() {
     # Check the interface parameter
-    if [ -z "${OCF_RESKEY_iface}" ]; then 
-        ocf_log err "Empty iface parameter.  Please specify network interface to check"
-        exit $OCF_ERR_CONFIGURED
+    if [ -z "${OCF_RESKEY_iface}" ]; then
+        if [ -z "${OCF_RESKEY_ip}" ]; then
+            ocf_log err "Empty iface and ip parameters. Please specify either an interface name or valid ip address."
+            exit $OCF_ERR_CONFIGURED
+        else
+            ipcheck_ipv4 "${OCF_RESKEY_ip}"
+            if [ $? = 1 ] ; then
```

Comment:
Imho it is better to use numeric comparison operators in such tests.
like [ $? -eq 1 ]

Reply:
I will change that.